scala/private/dependency.bzl

```diff
@@ -0,0 +1,78 @@
+# This file contains all computations for what the dependency mode is
+# (i.e. transitive, plus-one, direct, etc)
+# and what/how dependency analysis is performed (unused deps, strict deps, etc).
+
+def new_dependency_info(
+        dependency_mode,
+        unused_deps_mode,
+        strict_deps_mode,
+        dependency_tracking_method):
+    is_strict_deps_on = strict_deps_mode != "off"
+
+    # Currently we do not support running both strict and unused deps at the same time
+    if is_strict_deps_on:
+        unused_deps_mode = "off"
+
+    is_unused_deps_on = unused_deps_mode != "off"
+
+    need_direct_jars = is_strict_deps_on or is_unused_deps_on
+    need_direct_targets = is_unused_deps_on
+
+    return struct(
+        dependency_mode = dependency_mode,
+        need_indirect_info = is_strict_deps_on,
+        need_direct_jars = need_direct_jars,
+        need_direct_targets = need_direct_targets,
+        need_direct_info = need_direct_jars or need_direct_targets,
+        dependency_tracking_method = "high-level",
+        unused_deps_mode = unused_deps_mode,
+        strict_deps_mode = strict_deps_mode,
+        use_analyzer = is_strict_deps_on or is_unused_deps_on,
+    )
+
+def dependency_info_for_addons(ctx):
```

Comment:
This refers to protobuf and scrooge as things not part of "core" scala. Is there a term for that, if not I'll maybe think of some alternative terms.

Reply:
I think there's value in understanding why they have variance (or more importantly do we think they should have variance).
Maybe they shouldn't have and then we don't need a better name we just need to align it with the other. Maybe it's because they are generated code rules and then that's the name.
Maybe it's something else, not sure

Reply:
As per your other comment will file a ticket and will also rename to legacy_unclear_something or other